lib/jekyll-import/importers/drupal_common.rb

```diff
@@ -0,0 +1,148 @@
+require 'date'
+
+module JekyllImport
+  module Importers
+    module DrupalCommon
+      # This module provides a base for the Drupal importers (at least for 6
+      # and 7; since 8 will be a different beast). Version-specific importers
+      # will need to implement the missing methods from the Importer class.
+      #
+      # The general idea is that this importer reads a MySQL database via Sequel
+      # and creates a post file for each node it finds in the Drupal database.
+
+      module ClassMethods
+        def specify_options(c)
+          c.option 'dbname', '--dbname DB', 'Database name'
+          c.option 'user', '--user USER', 'Database user name'
+          c.option 'password', '--password PW', "Database user's password (default: '')"
+          c.option 'host', '--host HOST', 'Database host name (default: "localhost")'
+          c.option 'prefix', '--prefix PREFIX', 'Table prefix name'
+          c.option 'types', '--types TYPE1[,TYPE2[,TYPE3...]]', Array, 'The Drupal content types to be imported.'
+        end
+
+        def require_deps
+          JekyllImport.require_with_fallback(%w[
+          rubygems
+          sequel
+          fileutils
+          safe_yaml
+        ])
+        end
+
+        def process(options)
+          dbname = options.fetch('dbname')
+          user   = options.fetch('user')
+          pass   = options.fetch('password', '')
+          host   = options.fetch('host', 'localhost')
+          prefix = options.fetch('prefix', '')
+          types  = options.fetch('types', %w(blog story article))
+
+          db = Sequel.mysql(dbname, :user => user, :password => pass, :host => host, :encoding => 'utf8')
+
+          query = self.get_query(prefix, types)
+
+          src_dir = Jekyll.configuration({})['source']
+
+          dirs = {
+              :_posts => File.join(src_dir, '_posts').to_s,
+              :_drafts => File.join(src_dir, '_drafts').to_s,
+              :_layouts => File.join(src_dir, '_layouts').to_s
```

Comment:
If you're checking the configuration, it would be fruitful to check the value of `layouts_dir` here. It should default to `<source>/_layouts` but can be modified by the user.